packages/netlify-cms-core/src/actions/entries.ts

```diff
@@ -469,59 +454,94 @@ export function traverseCollectionCursor(collection, action) {
   };
 }
 
-export function createEmptyDraft(collection) {
-  return dispatch => {
+export function createEmptyDraft(collection: Collection) {
+  return async (dispatch: ThunkDispatch<State, {}, AnyAction>, getState: () => State) => {
     const dataFields = createEmptyDraftData(collection.get('fields', List()));
-    const newEntry = createEntry(collection.get('name'), '', '', { data: dataFields });
+
+    let mediaFiles = [] as MediaFile[];
+    if (!collection.has('media_folder')) {
+      await waitForMediaLibraryToLoad(dispatch, getState());
+      mediaFiles = getState().mediaLibrary.get('files');
+    }
+
+    const newEntry = createEntry(collection.get('name'), '', '', { data: dataFields, mediaFiles });
```

Comment:
Not sure why the `mediaFiles` are added here for a new entry, won't the files be persisted on save for editorial workflow mode even if the new entry didn't make use of the files?

Reply:
For an empty draft for a collection with a specific media folder the media files should be an empty array.
For an empty draft with for a collection no specific media folder the medial files should be populated from the "global" media library.

The `selectMediaFiles` selector detects if the user opened the media library from the editor or from the main toolbar and selects the draft media files or the "global" media files.

Another option will to check the collection `media_folder` property when selecting media files, but it makes sense for the draft entry state to contain the relevant media files for that entry.

Reply:
I mean if creating a new draft for a collection without  `media_folder` option, the global media files get loaded and added to the entry. Without inserting any media file in the draft, this line https://github.com/netlify/netlify-cms/pull/2958/files#diff-f54d023791c4074f77a150bb2ec73579R374 will select all the media files and persist them together with the draft, no?

Reply:
Only draft media files will be passed to the backend here:
https://github.com/netlify/netlify-cms/pull/2958/files#diff-72d48780a51d0d88f90945366a4ff84fR536

Reply:
Ok, got it.